### PR TITLE
Beef up resiliency of DeclareAsNullable fixer

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/64515

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@Cosifne could you ptal?  Thanks! :)

**Comment by CyrusNajmabadi** [comment]
@ryzngard ptal.  thanks!

**Comment by genlu** [comment]
Could you pls add a test? Thanks!

**Review by ryzngard — APPROVED** [review]
Sign off with expectation of test being added

**Comment by CyrusNajmabadi** [comment]
Don't have a test here.  We don't know what's causing this.  But we can see a lot of code here that is just not written in a resilient fashion to error recovery.  So this is just about putting this code into a healthy state where we shoudl be safe from this :)

File: src/Analyzers/CSharp/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs [hunk 1]
```diff
@@ -191,72 +180,49 @@ SyntaxKind.IndexerDeclaration or
...
-                    }
-                    else if (syntax is TupleElementSyntax tupleElement)
-                    {
+
+                    if (syntax is TupleElementSyntax tupleElement)
                         return tupleElement.Type;
-                    }
                 }
-                else if (symbol is IFieldSymbol { CorrespondingTupleField: IFieldSymbol tupleField })
+                else if (symbol is IFieldSymbol { CorrespondingTupleField: IFieldSymbol { Locations: [{ IsInSource: true } location] } })
```
Comment (genlu) [inline src/Analyzers/CSharp/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs:213]: Neat!

**Review by genlu — APPROVED** [review]

**Comment by ryzngard** [comment]
> Don't have a test here. We don't know what's causing this. But we can see a lot of code here that is just not written in a resilient fashion to error recovery. So this is just about putting this code into a healthy state where we shoudl be safe from this :)

I thought https://github.com/dotnet/roslyn/issues/64515#issuecomment-1271330479 had repro code?

**Comment by genlu** [comment]
Customer provided a code snippet in the linked issue. It doesn't trigger the error?

**Comment by CyrusNajmabadi** [comment]
wat.  i'm so confused :(  i swear i didn't see that :)  will add.

**Comment by CyrusNajmabadi** [comment]
So the user's code does not repro the issue for me :(

**Pull request merged**